core/src/main/scala/Graph.scala

```diff
@@ -903,6 +903,30 @@ case class Graph[N,A,B](rep: GraphRep[N,A,B]) {
     getLPath(t, lbft(s))
 
   /**
+   * Cheapest path from vertex `s` to vertex `t` under the cost function `costFkt` with labels
+   * @group bfs   
+   */
+  def cheapestPath[C](s: N, t: N, costFkt: (LNode[N,A],B,LNode[N,A]) => C)(implicit num: Numeric[C]): Option[LPath[N,B]] = {
```

Comment:
This could be a little more generic as `C: Monoid: Order`.  I imagine `C` would _usually_ be numeric in the real world, but it doesn't have to be.